File: Left array rotations.js
// baked in js Rhino

function leftrotatebyone(a){
    let temp = a[0]
    for(let i = 0; i < a.length - 1; i++)
        a[i] = a[i+1]
    a[a.length - 1] = temp
}


function leftarrayrotations(a, d){
    for(let i = 0;i < d; i++)
        leftrotatebyone(a)
}


a = [1, 2, 3, 4, 5, 6, 7]
leftarrayrotations(a, 2)
print(a)

File: left array rotations.py
# cook your dish here

def leftrotatebyone(a):
    temp = a[0]
    for i in range(len(a) - 1):
        a[i] = a[i+1]
    a[len(a) - 1] = temp

def leftrotate(a, d):
    for i in range(d):
        leftrotatebyone(a)

a = [1, 2, 3, 4, 5, 6, 7]
leftrotate(a, 2)
print(a)
